src/Api/AdminConsole/Controllers/OrganizationsController.cs

```diff
@@ -74,7 +76,8 @@ public OrganizationsController(
         IPushNotificationService pushNotificationService,
         IProviderRepository providerRepository,
         IProviderBillingService providerBillingService,
-        IDataProtectorTokenFactory<OrgDeleteTokenable> orgDeleteTokenDataFactory)
+        IDataProtectorTokenFactory<OrgDeleteTokenable> orgDeleteTokenDataFactory,
+        IRemoveOrganizationUserCommand removeOrganizationUserCommand)
```

Comment:
@withinfocus & @JaredSnider-Bitwarden I think it would be better to switch to action injection instead of constructor injection. We could save on memory consumption for every incoming request.

https://learn.microsoft.com/en-us/aspnet/core/mvc/controllers/dependency-injection?view=aspnetcore-8.0#action-injection-with-fromservices

You can inject it directly.

```csharp
public async Task Leave(Guid id, [FromServices] IRemoveOrganizationUserCommand removeOrganizationUserCommand)
```

That's 32 bytes saved at minimum for every other action in `OrganizationsController`

Reply:
@jonashendrickx , that's pretty neat! I wasn't aware of that, thanks! I'll wait for @withinfocus  to decide whether we want to use it, as it breaks our usual pattern.

Reply:
My opinion on the matter:

For controllers, it's mostly stylistic, @jonashendrickx is right there are a small amount of memory savings for all other actions if the service is transient or scoped and hasn't already been resolved. But I would never make this choice for those savings. If you care that much, don't use a controller action to begin with, start using minimal API's. But then speaking of minimal API's, they are likely a thing that will get used more and more in Bitwarden code and they will use method injected services like this so it's not a bad idea to get people used to it.

Reply:
Thanks for the input. I'll leave this as is for now to keep things consistent with our current pattern. We can revisit this if we move towards minimal APIs more broadly.